### DISCUSSION TITLE
bug: etcd cache

### DISCUSSION THREAD

**Discussion by yocaning** [apache/apisix#5241] [Q&A]
### Issue description

if you have own etcd cache like this：
    authorizations_etcd, error = core.config.new("/authorizations", {
        automatic = true,
        item_schema = app_scheme
    })

if the key authorizations dont have a key value （init_dir）,the etcd cache dont load when apisix restart 

### Environment

- apisix version (cmd: `apisix version`):
- OS (cmd: `uname -a`):
- OpenResty / Nginx version (cmd: `nginx -V` or `openresty -V`):
- etcd version, if have (cmd: run `curl http://127.0.0.1:9090/v1/server_info` to get the info from server-info API):
- apisix-dashboard version, if have:
- the plugin runner version, if the issue is about a plugin runner (cmd: depended on the kind of runner):
- luarocks version, if the issue is about installation (cmd: `luarocks --version`):

### Steps to reproduce

1.authorizations_etcd, error = core.config.new("/authorizations", {
        automatic = true,
        item_schema = app_scheme
    })
2.restart apisix
3. the authorizations_etcd dont have data

### Actual result

when restart apisix.   load the etcd cache data

### Error log

--- dont have  nodes array
{"key":"\/gateway\/authorizations\/AKIDz8krbsJ5yKBZQpn74WFkmLPx3EXAMPLE1","createdIndex":26295,"modifiedIndex":26473,"value":{"secret":"AKIDz8krbsJ5yKBZQpn74WFkmLPx3EXAMPLE11","username":"yyy"}}

---- --other key ex:routes:  (hava nodes array)
{"key":"\/gateway\/routes","nodes":[{"key":"\/ioa-gateway\/routes\/00000000000000025212","createdIndex":25213,"modifiedIndex":25247,"value":{"update_time":1634119249,"methods":["GET","POST"],"id":"00000000000000025212","upstream":{"type":"roundrobin","hash_on":"vars","nodes":{"9.135:11111":1},"pass_host":"pass"},"create_time":1634118070,"status":1,"plugins":{"cors":{"allow_credential":false,"allow_origins":"*","max_age":5,"expose_headers":"*","allow_methods":"*","allow_headers":"*"}}

**Comment by tzssangglass** [comment]
Is this the code you implemented yourself?  If so, I'll turn this issue into a discussion and we'll discuss it in the discussion.

**Comment by yocaning** [comment]
> Is this the code you implemented yourself? If so, I'll turn this issue into a discussion and we'll discuss it in the discussion.

yes

**Reply by tzssangglass** [reply]
You can look at the implementation of sync etcd data in APISIX, for example: https://github.com/apache/apisix/blob/0e7ff54e6e7ef44bee85d9f38752486d9c104921/apisix/plugin.lua# L490-L498.

In short, sync etcd data is at the `init_worker_by_lua`, which is not easy, but it can be done.
